src/java/org/apache/cassandra/sensors/RequestSensorsFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sensors;
+
+import java.util.Optional;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.utils.FBUtilities;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.REQUEST_SENSORS_FACTORY;
+
+/**
+ * Provides a customizable factory to create a {@link RequestSensors} to track sensors per user request.
+ */
+
+public interface RequestSensorsFactory
```

Comment:
I see a couple more invocation of `RequestTracker.instance.get()` in SystemKeyspace, could you replace those as well?

Reply:
The `SystemKeysapce` use case is different. It should get the `RequestSensors` from the executor local but never create one itself. Only verb handler use `RequestSensorsFactory` strictly when the user request life cycle begins.  See `[Commit|Prepare|Propose]VerbHandler]` LWT handlers. 

It is possible push down `RequestSensors` from the handler to the SystemKeyspace, but accessing the `ExecutorLocal` keeps the method signatures clean.